src/libstd/ffi/c_str.rs

```diff
@@ -288,6 +288,26 @@ impl CString {
     /// Failure to call [`from_raw`] will lead to a memory leak.
     ///
     /// [`from_raw`]: #method.from_raw
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::ffi::CString;
+    ///
+    /// let c_string = CString::new("foo").unwrap();
+    ///
+    /// let ptr = c_string.into_raw();
+    ///
+    /// unsafe {
+    ///     assert_eq!(b'f', *ptr as u8);
+    ///     assert_eq!(b'o', *ptr.offset(1) as u8);
+    ///     assert_eq!(b'o', *ptr.offset(2) as u8);
+    ///     assert_eq!(b'\0', *ptr.offset(3) as u8);
```

Comment:
I'm less certain about this one. If people are used to working with raw pointers in Rust this is fine, but i wonder if we can do some other kind of comparison to not have to introduce `offset` in these docs. I don't anticipate people looking at CString with a solid understanding of Rust pointers. This is frequently an entry point into FFI, so readers may potentially only have a cursory understanding of pointers.